dev/io.openliberty.io.smallrye.config/src/io/openliberty/microprofile/config/internal/extension/OLSmallRyeConfigExtension.java

```diff
@@ -10,15 +10,59 @@
  *******************************************************************************/
 package io.openliberty.microprofile.config.internal.extension;
 
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.AfterDeploymentValidation;
 import javax.enterprise.inject.spi.Extension;
 
 import org.osgi.service.component.annotations.Component;
 import org.osgi.service.component.annotations.ConfigurationPolicy;
 
 import com.ibm.ws.cdi.extension.WebSphereCDIExtension;
 
+import io.openliberty.checkpoint.spi.CheckpointPhase;
 import io.smallrye.config.inject.ConfigExtension;
 
 @Component(service = WebSphereCDIExtension.class, configurationPolicy = ConfigurationPolicy.IGNORE, property = { "service.vendor=IBM" }, immediate = true)
 public class OLSmallRyeConfigExtension extends ConfigExtension implements Extension, WebSphereCDIExtension {
+    static final CheckpointPhase phase = CheckpointPhase.getPhase();
+    static final ThreadLocal<AtomicBoolean> recordingStopped = new ThreadLocal<AtomicBoolean>() {
+        @Override
+        protected AtomicBoolean initialValue() {
+            return new AtomicBoolean();
+        }
+    };
+
+    @Override
+    protected void validate(@Observes AfterDeploymentValidation adv) {
+        boolean stopped = stopRecordingReads();
+        try {
+            super.validate(adv);
+        } finally {
+            startRecordingReads(stopped);
+        }
+    }
+
+    static void startRecordingReads(boolean stopped) {
+        if (!stopped || phase == null || phase.restored()) {
+            return;
+        }
```

Comment:
Having to check whether `phase` is `null` everywhere is unfortunate. Not sure whether the design of `CheckpointPhase.phase()` could be changed there.

Reply:
Originally I wanted `CheckpointPhase.phase()` to return an `Optional<CheckpointPhase>`, but as of now that project is still compiled to Java 7 because it contains the code that reports that Java 7 is not supported when trying to run on Java 7.

We could consider making `CheckpointPhase.phase()` never return null, but then we are in the awkward spot of figuring out the enum constant name for the `CheckpointPhase` that means no checkpoint is actually going on.

Reply:
I guess it depends what else you're doing with `CheckpointPhase` elsewhere. Here it seems like it would be easy to have a phase named `INACTIVE` where `.restored()` always returns `true`, but I don't know what other considerations you have elsewhere in the code.

I did have a brief look in `CheckpointPhase` and it sounded like the phase may initially be `null` and then change to an actual value so it sounds like `null` can either be "we're not checkpointing" or "we haven't started thinking about checkpointing yet"?

Reply:
The intention is that `CheckpointPhase.phase()` will return a constant value.  Once it rerturns null it must always return null.  Down in boot strap I do see a window where that may not be true.  The code probably should throw an IllegalStateException from `phase()` if `CheckpointPhase.phaseSet==false`.

I can look at changing that to return an INACTIVE enum.  But I don't want to hold up this PR waiting for that to happen.